sale_weekday_delivery/models/res_partner.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+import calendar
+from datetime import timedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+WEEKDAY_MAPPING = {
+    "0": "monday",
+    "1": "tuesday",
+    "2": "wednesday",
+    "3": "thursday",
+    "4": "friday",
+    "5": "saturday",
+    "6": "sunday",
+}
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    delivery_schedule = fields.Selection(
+        [("direct", "As soon as possible"), ("fix_weekdays", "Fixed week days")],
+        string="Delivery schedule preference",
+        default="direct",
+        required=True,
+        help="Define the scheduling preference for delivery orders:\n\n"
+        "* As soon as possible: Do not postpone deliveries\n"
+        "* Fixed week days: Postpone deliveries to the next preferred "
+        "weekday",
+    )
+
+    delivery_schedule_monday = fields.Boolean(default=True, string="Monday")
+    delivery_schedule_tuesday = fields.Boolean(default=True, string="Tuesday")
+    delivery_schedule_wednesday = fields.Boolean(default=True, string="Wednesday")
+    delivery_schedule_thursday = fields.Boolean(default=True, string="Thursday")
+    delivery_schedule_friday = fields.Boolean(default=True, string="Friday")
+    delivery_schedule_saturday = fields.Boolean(default=True, string="Saturday")
+    delivery_schedule_sunday = fields.Boolean(default=True, string="Sunday")
+
+    @api.constrains(
+        "delivery_schedule",
+        "delivery_schedule_monday",
+        "delivery_schedule_tuesday",
+        "delivery_schedule_wednesday",
+        "delivery_schedule_thursday",
+        "delivery_schedule_friday",
+        "delivery_schedule_saturday",
+        "delivery_schedule_sunday",
+    )
+    def _check_delivery_schedule(self):
+        """Ensure at least one preferred day is defined"""
+        days_fields = ["delivery_schedule_%s" % d.lower() for d in calendar.day_name]
+        for partner in self:
+            if partner.delivery_schedule == "fix_weekdays" and not any(
+                [partner[d] for d in days_fields]
+            ):
+                raise ValidationError(
+                    _(
+                        "At least one preferred weekday must be defined if "
+                        "partner is set as 'Fixed week days' deliveries."
+                    )
+                )
+
+    def get_delivery_schedule_preferred_weekdays(self):
+        """List preferred weekdays"""
+        self.ensure_one()
```

Comment:
you can get rid of this and use the serialized field

Reply:
I'm not in favor as discussed with @grindtildeath

Reply:
that's fine, thinking about it twice it can stay and use the new fields anyway.
@jgrandguillaume what's the purpose of storing all those fields?
To me it makes no sense. Also, you are hacking everywhere you need to read those values...

Reply:
@simahawk we're going to rework this anyway, cf discussion below :wink: